Classes/Search/GithubClient+Search.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  GithubClient+Search.swift
+//  Freetime
+//
+//  Created by Sherlock, James on 28/07/2017.
+//  Copyright © 2017 Ryan Nystrom. All rights reserved.
+//
+
+import Foundation
+
+extension GithubClient {
+    
+    enum SearchResultType {
+        case error
+        case success(String?, [SearchResult])
+    }
+    
+    func search(query: String, before: String? = nil, containerWidth: CGFloat, completion: @escaping (SearchResultType) -> ()) {
+        let query = SearchReposQuery(search: query, before: before)
+        
+        apollo.fetch(query: query, cachePolicy: .fetchIgnoringCacheData) { (result, error) in
+            guard error == nil, result?.errors == nil else {
+                ShowErrorStatusBar(graphQLErrors: result?.errors, networkError: error)
+                return
```

Comment:
Admittedly had similar questions when doing this, but for the most part was copied and modified versions of the Notifications fetching - Error handling is probably something that could be handled better throughout the app as it is.

I'll take a look into this specifically though and see what improvements can be made